src/never.rs

```diff
@@ -0,0 +1,91 @@
+use {Future, Poll};
+
+/// A future which is never resolved.
+///
+/// This future can be created with the `empty` function.
+#[derive(Copy, Clone)]
+pub struct Never {}
+
+/// Creates a future which never resolves, representing a computation that never
+/// finishes.
+///
+/// The returned future will never resolve with a success but is still
+/// susceptible to cancellation. That is, if a callback is scheduled on the
+/// returned future, it is only run once the future is dropped (canceled).
+pub fn never<T, E>() -> impl Future<Item = T, Error = E> + Send + 'static {
```

Comment:
Default type parameters have been stuck for some time now because they raise a lot of contradictions. The simplest version of explaining the problem is that you can easily run into conflicting defaults and that makes them kind of a dangerous tool for composability (adding defaults is not possible in a backwards compatible way; and if a lot of fns have defaults then these defaults are likely to conflict and so they won't be of much use). We've been wrestling with various possible solutions to this problem -- some ideas just rule out default type parameters on fns altogether, iirc. Here is a comment I wrote and some ensuing discussion that may shed more light: https://github.com/rust-lang/rust/issues/27336#issuecomment-229988042